src/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -378,68 +415,115 @@ function compute_pressure!(boundary_model, ::Union{PressureMirroring, PressureZe
     return boundary_model
 end
 
-@inline function adami_pressure_extrapolation!(boundary_model, system,
-                                               neighbor_system::FluidSystem,
-                                               system_coords, neighbor_coords,
-                                               v_neighbor_system, neighborhood_search)
-    (; pressure, cache, viscosity) = boundary_model
+@inline function bnd_pressure_extrapolation!(boundary_model, system,
+                                             neighbor_system::FluidSystem,
+                                             system_coords, neighbor_coords, v,
+                                             v_neighbor_system, neighborhood_search)
+    (; pressure, cache, viscosity, density_calculator) = boundary_model
+    (; pressure_offset) = density_calculator
 
     # Loop over all pairs of particles and neighbors within the kernel cutoff.
     foreach_point_neighbor(system, neighbor_system, system_coords, neighbor_coords,
                            neighborhood_search;
                            points=eachparticle(system)) do particle, neighbor,
                                                            pos_diff, distance
-        adami_pressure_inner!(boundary_model, system, neighbor_system,
-                              v_neighbor_system, particle, neighbor, pos_diff,
-                              distance, viscosity, cache, pressure)
+        bnd_pressure_inner!(boundary_model, density_calculator, system, neighbor_system, v,
+                            v_neighbor_system, particle, neighbor, pos_diff,
+                            distance, viscosity, cache, pressure, pressure_offset)
     end
 end
 
-@inline function adami_pressure_inner!(boundary_model, system, neighbor_system,
-                                       v_neighbor_system, particle, neighbor, pos_diff,
-                                       distance, viscosity, cache, pressure)
+@inline function bnd_pressure_inner!(boundary_model, ::BernoulliPressureExtrapolation,
+                                     system, neighbor_system::FluidSystem, v,
+                                     v_neighbor_system, particle, neighbor, pos_diff,
+                                     distance, viscosity, cache, pressure,
+                                     pressure_offset)
+    density_neighbor = particle_density(v_neighbor_system, neighbor_system, neighbor)
+
+    resulting_acc = neighbor_system.acceleration - current_acceleration(system, particle)
+
+    kernel_weight = smoothing_kernel(boundary_model, distance)
+
+    pressure[particle] += (pressure_offset +
+                           particle_pressure(v_neighbor_system, neighbor_system,
+                                             neighbor) +
+                           dynamic_pressure(density_neighbor, v, v_neighbor_system,
+                                            particle, neighbor, system)
```

Comment:
Can't you just add this in the Adami function below and pass the density calculator to `dynamic_pressure`, so that you can dispatch and return zero for Adami?